utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,452 @@
+            pooling_mode=pooling_mode,
+        )
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception during making model config file for {model_format} model: {e}"
+
+    # 4.) Return model_path & model_config_path for model registration
+    model_config_path = folder_path + MODEL_CONFIG_FILE_NAME
+    return model_path, model_config_path
+
+
+def register_and_deploy_sentence_transformer_model(
+    ml_client: "MLCommonClient",
+    model_path: str,
+    model_config_path: str,
+    model_format: str,
+) -> List["DTypeLike"]:
+    """
+    Register the pretrained sentence transformer model by using the model file and the model config file,
+    deploy the model to generate embeddings for the TEST_SENTENCES,
+    and return the embeddings for model verification
+
+    :param ml_client: A client that communicates to the ml-common plugin for OpenSearch
+    :type ml_client: MLCommonClient
+    :param model_path: Path to model file
+    :type model_path: string
+    :param model_config_path: Path to model config file
+    :type model_config_path: string
+    :param model_format: Model format ("TORCH_SCRIPT" or "ONNX")
+    :type model_format: string
+    """
+    embedding_data = None
+
+    # 1.) Register & Deploy the model
+    model_id = ""
+    try:
+        model_id = ml_client.register_model(
+            model_path=model_path,
+            model_config_path=model_config_path,
+            deploy_model=True,
+            isVerbose=True,
+        )
+        print()
+        print(f"{model_format}_model_id:", model_id)
+        assert model_id != "" or model_id is not None
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception in {model_format} model registration/deployment: {e}"
+
+    # 2.) Check model status
+    try:
+        ml_model_status = ml_client.get_model_info(model_id)
+        print()
+        print("Model Status:")
+        print(ml_model_status)
+        assert ml_model_status.get("model_format") == model_format
+        assert ml_model_status.get("algorithm") == "TEXT_EMBEDDING"
+    except Exception as e:
+        assert False, f"Raised Exception in getting {model_format} model info: {e}"
+
+    # 3.) Generate embeddings
+    try:
+        embedding_output = ml_client.generate_embedding(model_id, TEST_SENTENCES)
+        assert len(embedding_output.get("inference_results")) == 2
+        embedding_data = [
+            embedding_output["inference_results"][i]["output"][0]["data"]
+            for i in range(len(TEST_SENTENCES))
+        ]
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception in generating sentence embedding with {model_format} model: {e}"
+
+    # 4.) Undeploy the model
+    try:
+        ml_client.undeploy_model(model_id)
+        ml_model_status = ml_client.get_model_info(model_id)
+        assert ml_model_status.get("model_state") != "UNDEPLOY_FAILED"
```

Comment:
What other status are we getting here? Can't we test if the model status is "undeployed" or not?

Reply:
Let me look into that. I follow the integration test here.